src/SFA.DAS.EAS.Portal/Application/Services/IMessageContext.cs

```diff
@@ -1,10 +1,13 @@
 ﻿using System;
+using NServiceBus;
 
 namespace SFA.DAS.EAS.Portal.Application.Services
 {
     public interface IMessageContext
     {
         string Id { get; set; }
         DateTime CreatedDateTime { get; set; }
+
+        void Initialise(IMessageHandlerContext handlerContext);
```

Comment:
It doesn't make sense for Intialise with a IMessageHandlerContext to be on this interface.  I understand why you have (to remove the extension method).  To achieve this it would be better design to create another interface for this method that MessageContext implements.  This interface design is intended for the consumers later in the flow they have no dependency on IMessageHandlerContext.

Reply:
Good point. I've taken `Initialise` out of `IMessageContext` and moved it into a new `IMessageContextInitialisation` interface.